### DISCUSSION TITLE
Data importer fails to import a transaction with amount (in foreign currency) and currency code

### DISCUSSION THREAD

**Discussion by citxx** [firefly-iii/firefly-iii#8809] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Data importer fails to import a transaction on the "data submission" step with the following error:

```
Line #0: transactions.0.foreign_amount: The content of this field is invalid without currency information. (original value: "8.000000000000")
```

This happens while both `Amount (in foreign currency)` and `Currency code (ISO 4217)` fields are present in CSV and correctly assigned in data importer. 

### Debug information

Debug information generated at 2024-04-22 23:56:50 for Firefly III version **v6.1.13**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>6.1.13 / v2.0.13 / 23 (exp. 23)</td></tr><tr><td>PHP version</td><td>8.3.4 (64bits) / apache2handler / Linux x86_64</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, pgsql, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>1052, base <span>#</span>82</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>Europe/Berlin + Europe/Berlin</td></tr><tr><td>App environment</td><td>production, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>notice, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>en_US + equal</td></tr><tr><td>Trusted proxies</td><td>**</td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>2024-04-21 22:00:00 (23 hours ago)</td></tr><tr><td>Mailer</td><td>log</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td><span title="Has rules">:wrench:</span> <span title="Has object groups">:bookmark_tabs:</span> <span title="Has subscriptions">:email:</span></td></tr><tr><td>Session start</td><td>2024-01-01 00:00:00</td></tr><tr><td>Session end</td><td>2024-04-22 22:59:48</td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>en_US</td></tr><tr><td>User locale</td><td>en_US</td></tr><tr><td>Locale(s) supported</td><td>en_US.utf8: :white_check_mark:<br>en_US.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36</td></tr></tbody></table>

### Expected behaviour

The transaction should be imported.

### Steps to reproduce

I was able to reproduce the bug on the demo instance with demo importer as well:

1. Open https://data-importer.firefly-iii.org/
2. Use `https://demo.firefly-iii.org/` as URL and `4` as Client ID
3. `Import file`, then use [transactions-example.csv](https://github.com/firefly-iii/firefly-iii/files/15068971/transactions-example.csv) and [config-example.json](https://github.com/firefly-iii/firefly-iii/files/15069073/config-example.json)
4. `Submit` until you see the `Map data` screen
5. Chose `Hungarian forint` as the mapped value for the only currency code.
6. Proceed with import

The problem is still present without mapping, but then `THB` should be replaced in CSV file with a currency existing in the demo instance (e.g. `HUF`).

### Additional info

I hope it's not a stupid mistake by me, that I wasn't able to find after checking multiple times 😅

Relevant importer logs:

```
[2024-04-22 23:29:31] local.INFO: Now in App\Services\CSV\Conversion\Routine\CSVFileProcessor::processCSVLines with 1 records  
[2024-04-22 23:29:31] local.INFO: Parsed all 1 lines.  
[2024-04-22 23:29:31] local.INFO: Going to remove duplicate lines.  
[2024-04-22 23:29:31] local.INFO: Went from 1 line(s) to 1 line(s)  
[2024-04-22 23:29:31] local.INFO: Now processing the data in the 1 CSV lines...  
[2024-04-22 23:29:31] local.INFO: Done processing data in 1 CSV lines...  
[2024-04-22 23:29:31] local.INFO: Now parsing and combining 1 lines.  
[2024-04-22 23:29:31] local.INFO: Done parsing and combining 1 lines.  
[2024-04-22 23:29:31] local.INFO: Converting 1 line(s) into transactions.  
[2024-04-22 23:29:31] local.INFO: Now processing line 1/1.  
[2024-04-22 23:29:31] local.INFO: Done converting 1 line(s) into transactions.  
...
[2024-04-22 23:29:41] local.INFO: Going to submit 1 transactions to your Firefly III instance.  
[2024-04-22 23:29:41] local.ERROR: Submission error: 0 ["The content of this field is invalid without currency information."] 
[2024-04-22 23:29:41] local.ERROR: Add error to index #0: transactions.0.foreign_amount: The content of this field is invalid without currency information. (original value: "8.000000000000")  
[2024-04-22 23:29:41] local.ERROR: transactions.0.foreign_amount: The content of this field is invalid without currency information. (original value: "8.000000000000")  
[2024-04-22 23:29:41] local.INFO: Done submitting 1 transactions to your Firefly III instance.  
```

**Comment by JC5** [comment, marked as answer]
Hi,

Please select "Foreign currency code (ISO 4217)", that should do the trick.

**Reply by citxx** [reply]
Thanks for the help! It was a stupid mistake in the end :)

It might be easier for new users to avoid this mistake if "Currency code" and "Foreign currency code" were next to each other in the dropdown.
